compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1840,7 +1871,10 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         });
         for candidate in candidates_to_expand.iter_mut() {
             if !candidate.subcandidates.is_empty() {
-                self.merge_trivial_subcandidates(candidate);
+                // FIXME: Support merging trival candidates in branch coverage instrumentation
```

Comment:
This means that turning on branch coverage can cause exponential blowups in MIR size, this is unfortunate. Are you sure tracking or-patterns is worth it? (I'm not expert here, just curious)